Off-chain workers docs (#379)
substrate-developer-hub/substrate-developer-hub.github.io: docs/conceptual/core/off-chain-workers.md

To make the off-chain data integration secure and more efficient, Substrate provides off-chain workers. The off-chain worker subsystem allows execution of long-running and possibly non-deterministic tasks (e.g. web requests, encryption/decryption and signing of data, random number generation, CPU-intensive computations, enumeration/aggregation of on-chain data, etc.) which could otherwise require longer than the block execution time.

Off-chain workers have their own Wasm execution environment outside of the Substrate runtime. This separation of concerns is to make sure that the block production is not impacted by the long-running tasks. However, as the off-chain workers are declared in the same code as the runtime, they can easily access on-chain state for their computations.

![Off-chain Workers](/docs/assets/off-chain-workers-02.png)

## APIs

> Additionally, they have these capabilities to communicate with the external world via extended set of APIs.

Reviewer: ```suggestion
Offchain workers have access to extended APIs for communicating with the external world:
```

* Ability to submit transactions (either signed or unsigned) back on-chain to publish computation results.
* A fully-featured HTTP client allowing to access and fetch data from external services.
* Access to the local keystore to sign and verify statements or transactions.
* An additional, local key-value database shared between all off-chain workers.

> * A secure, local entropy source for random number generation.

Reviewer: I almost removed the comma, but I think it's actually correct as it stands https://www.quickanddirtytips.com/education/grammar/commas-with-adjectives

> * Access to local node precise time and ability to sleep/resume work.

Reviewer: ```suggestion
* Access to the node's precise local time and the ability to sleep and resume work.
```

> Off-chain workers can be initiated from within a special function in your runtime implementation, `fn offchain_worker(block: T::BlockNumber)`. The function is executed after each block import. To communicate results back to the chain, off-chain workers can submit signed or unsigned transactions that are later to be included in subsequent blocks.

Reviewer: ```suggestion
Off-chain workers can be initiated from within a special function in your runtime implementation, `fn offchain_worker(block: T::BlockNumber)`. The function is executed after each block import. To communicate results back to the chain, off-chain workers can submit signed or unsigned transactions to be included in subsequent blocks.
```
